src/main/java/com/limechain/constants/GenesisBlockHash.java

```diff
@@ -1,16 +1,81 @@
-    public static final Hash256 POLKADOT =
-            Hash256.from("91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3");
-    public static final Hash256 WESTEND =
-            Hash256.from("e143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e");
-    public static final Hash256 KUSAMA =
-            Hash256.from("b0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe");
-    public static final Hash256 LOCAL =
-            Hash256.from("f825d3c387965817822a437528e6f4b659480b54d072c8d99b8af8451185c58f");
+    private final Hash256 genesisHash;
+    private final TrieStructure<NodeData> genesisTrie;
+    private final BlockHeader genesisBlockHeader;
+    private final ChainService chainService;
+
+    public GenesisBlockHash(ChainService chainService) {
+        this.chainService = chainService;
+        this.genesisTrie = buildGenesisTrie();
+        this.genesisBlockHeader = buildGenesisBlock(getMerkleValue(this.genesisTrie));
+//        this.genesisHash = this.genesisBlockHeader.getHash(); //TODO: Uncomment when fixed
+
+        //TODO: only temporary solution
+        Hash256 POLKADOT = Hash256.from("91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3");
+        Hash256 WESTEND = Hash256.from("e143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e");
+        Hash256 KUSAMA = Hash256.from("b0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe");
+        Hash256 LOCAL = Hash256.from("f825d3c387965817822a437528e6f4b659480b54d072c8d99b8af8451185c58f");
+
+        Network network = AppBean.getBean(Network.class);
+
+        switch (network.getChain()) {
+            case POLKADOT -> genesisHash = POLKADOT;
+            case KUSAMA -> genesisHash = KUSAMA;
+            case WESTEND -> genesisHash = WESTEND;
+            case LOCAL -> genesisHash = LOCAL;
+            default -> throw new IllegalStateException("Unexpected value: " + network.getChain());
+        }
+        //TODO!: only temporary solution
+    }
+
+    public BlockHeader buildGenesisBlock(byte[] merkleValue) {
+        BlockHeader blockHeader = new BlockHeader();
+        blockHeader.setParentHash(Hash256.empty());
+        blockHeader.setBlockNumber(BigInteger.ZERO);
+        blockHeader.setStateRoot(new Hash256(merkleValue));
+        blockHeader.setExtrinsicsRoot(Hash256.from(HexUtils.toHexString(HashUtils.hashWithBlake2b(new byte[1]))));
+        blockHeader.setDigest(new HeaderDigest[0]);
+
+        return blockHeader;
+    }
+
+    byte[] getMerkleValue(TrieStructure<NodeData> trieStructure) {
+        //TODO: GENESIS STATE ROOT MERKLE VALUE MUST BE
+        // '0x29d0d972cd27cbc511e9589fcb7a4506d5eb6a9e8df205f00472e5ab354a4e17' for the polkadot chain
+        return trieStructure.getRootNode()
+                .map(NodeHandle::getUserData)
+                .map(NodeData::getMerkleValue)
+                .orElseThrow(() -> new RuntimeException("Root node merkle value not found"));
+    }
```

Comment:
I don't understand this comment

Reply:
Currently, the gerMerkleValue function is returning the wrong value, because of the wrong implementation in the merkle value calculation. The 0x29d0... is the expected value, for the genesis files root merkle value calculation. I added it in this todo because we were initially unsure which value to expect.

Reply:
We're calculating the wrong root hash - something is not right, but we'll figure it out later. That's the issue we discussed this morning on the daily.